### ISSUE TITLE
Mod fails with Exception: System.TypeLoadException

### ISSUE THREAD

**Issue by XekriRedmane** [RileyGuy/Impressive#3]
Including logs.

Some highlights:

```
6:15:36 PM.852 ( -1 FPS)	Initializing App: Beta 2023.12.21.1127
Platform: Windows
HeadDevice: SteamVR
IsAOT: False
OS: Windows 10  (10.0.22621) 64bit
CPU: 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz
GPU: NVIDIA GeForce RTX 3080 Ti
PhysicalCores: 
MemoryBytes: 127.88 GB
VRAMBytes: 11.82 GB
MaxTextureSize: 16384
IsGPUTexturePOTByteAligned: True
UsingLinearSpace: True
XR Device Name: OpenVR
XR Device Model: Oculus Quest Pro
StereoRenderingMode: SinglePass
Max GC Generation: 0, IsLittleEndian: True
System.Numerics.Vectors HW accelerated: False, Vector<float>.Count: 4
Brotli native encoding/decoding supported: True
```

```
6:15:40 PM.329 ( -1 FPS)	[INFO] [ResoniteModLoader] Loading assemblies from rml_libs
6:15:40 PM.353 ( -1 FPS)	[INFO] [ResoniteModLoader] F:\Steam\steamapps\common\Resonite\Libraries\ResoniteModLoader.config is missing! This is probably fine.
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] Loaded libraries from rml_libs:
0Harmony, Version=2.2.2.0, Culture=neutral, PublicKeyToken=null Sha256=6928F117D52C0683C44683C0656CD5A345D21AA45720B11C64DC72A35771CAC5
Re-Sounding, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null Sha256=00407E18F55AE5285E514E010C9BB46472ED0E9FDF74A28C939C20FC567F7F74
Rug.Osc, Version=1.2.5.17694, Culture=neutral, PublicKeyToken=null Sha256=69691768F4E4042EE7631DFBAF2C26FD7B79F6134DAC71DAD7D0A9BBD8F6F955
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] ResoniteModLoader v2.4.0 starting up!
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] CLR v4.0.30319.42000
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] Using Harmony v2.2.2.0
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] Using Elements.Core v1.0.0.0
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] Using FrooxEngine v2023.12.21.1127
6:15:41 PM.283 ( -1 FPS)	[INFO] [ResoniteModLoader] Using Json.NET v13.0.0.0
6:15:41 PM.505 ( -1 FPS)	[INFO] [ResoniteModLoader] Compatibility hash spoofing succeeded
6:15:41 PM.506 ( -1 FPS)	[INFO] [ResoniteModLoader] Loading assemblies from rml_mods
6:15:42 PM.051 ( -1 FPS)	[INFO] [ResoniteModLoader] Loaded mod [Impressive/1.0.3.0] (Impressive.dll) by Cyro with Sha256: 56B42E28ECA0287F8298091AEDC0E4F846B716FF32D23FC667357AA2ED0EE831
6:15:42 PM.279 ( -1 FPS)	[INFO] [ResoniteModLoader] Loaded mod [InspectorScroll/2.0.0] (InspectorScroll.dll) by art0007i with Sha256: 2DBCEA4B00F4650C8B06A76D6DFD4636977329AB1E26239DD61187A03848EB24
6:15:42 PM.386 ( -1 FPS)	[INFO] [ResoniteModLoader/Impressive] Patched successfully!
6:15:42 PM.550 ( -1 FPS)	[ERROR][ResoniteModLoader] Exception in execution hook!
System.NullReferenceException: Null method for com.resonitemodloader
  at HarmonyLib.PatchProcessor.Patch () [0x0001d] in <2414a8eb44e8496080270436f777e942>:0 
  at HarmonyLib.Harmony.Patch (System.Reflection.MethodBase original, HarmonyLib.HarmonyMethod prefix, HarmonyLib.HarmonyMethod postfix, HarmonyLib.HarmonyMethod transpiler, HarmonyLib.HarmonyMethod finalizer) [0x00028] in <2414a8eb44e8496080270436f777e942>:0 
  at ResoniteModLoader.AssemblyHider.PatchResonite (HarmonyLib.Harmony harmony, System.Collections.Generic.HashSet`1[T] initialAssemblies) [0x000f1] in <e7e58f1fcd484402abeda23e92f7b020>:0 
  at ResoniteModLoader.HarmonyWorker.LoadModsAndHideModAssemblies (System.Collections.Generic.HashSet`1[T] initialAssemblies) [0x00010] in <e7e58f1fcd484402abeda23e92f7b020>:0 
  at ResoniteModLoader.ExecutionHook..cctor () [0x0005a] in <e7e58f1fcd484402abeda23e92f7b020>:0 
```

```
6:15:50 PM.260 ( -1 FPS)	[25] Initializing SessionAnnouncer...
6:15:50 PM.280 ( -1 FPS)	[INFO] [ResoniteModLoader/Impressive] Failed to initialize SteamLink driver! Exception: System.TypeLoadException: Could not resolve type with token 01000049 (from typeref, class/assembly OnChangedHandler, )
  at FrooxEngine.InputInterface.RegisterInputDriver (FrooxEngine.IInputDriver driver) [0x0001e] in <b95f6ac3437f4760b144032724fb496c>:0 
  at Impressive.Impressive+<>c__DisplayClass9_0.<OnEngineInit>b__0 () [0x00010] in <2d92bf30e7474896bd99c37c667217b0>:0 
6:15:50 PM.280 ( -1 FPS)	[26] FrooxEngine Initialization Finished
```

[DESKTOP-83DJ9R8 - 2023.12.21.1127 - 2023-12-28 18_15_29.log](https://github.com/RileyGuy/Impressive/files/13790514/DESKTOP-83DJ9R8.-.2023.12.21.1127.-.2023-12-28.18_15_29.log)

**Comment by XekriRedmane** [comment]
Resolved: Upgrading RML to 2.5.1 fixed the issue. Maybe a note in README.md to say RML >= 2.5.1 is required!

**Comment by BlueCyro** [comment]
Thanks, I'll definitely note this in the readme when I get a chance
